src/pykeen/pipeline/api.py

```diff
@@ -567,10 +646,19 @@ def save_pipeline_results_to_directory(
         )
         for epoch, loss in enumerate(pipeline_result.losses):
             losses_rows.append((i, epoch, loss))
+        result_comparator.parse_from_result(result=pipeline_result)
 
     losses_df = pd.DataFrame(losses_rows, columns=["Replicate", "Epoch", "Loss"])
     losses_df.to_csv(directory.joinpath("all_replicates_losses.tsv"), sep="\t", index=False)
 
+    if result_comparator.is_non_empty():
+        metric_df = result_comparator.get_df()
+        metric_df.to_csv(directory.joinpath("all_replicates_metrics.tsv"), sep="\t", index=False)
+        logger.debug(f"metric results: {metric_df}")
+
+        # summarize
+        logger.info(compare_results(metric_df).to_string())
```

Comment:
why not also write this dataframe?

Reply:
:shrug: I guess because it is derived? :sweat_smile:

Reply:
I think it's better to have it additionally written as a file in case a user wants to look back later without figuring out where the code is in PyKEEN

Reply:
Added in df21c1ef